CHANGELOG.md

```diff
@@ -36,6 +36,13 @@
 - Enforce naming conventions for Invalid[Trace|Span]ID: (#5969)
   - Deprecate funcs `pcommon.InvalidTraceID` and `pcommon.InvalidSpanID` in favor of vars `pcommon.EmptyTraceID` and `pcommon.EmptySpanID` 
 - Deprecate `Update` and `Upsert` methods of `pcommon.Map` (#5975)
+- Deprecated the current MetricDataPointFlags API.  The new API provides functions to check and set Flags. (#5999)
+  - `NumberDataPoint.Flags` -> `NumberDataPoint.FlagsImmutable`
+  - `HistogramDataPoint.Flags` -> `HistogramDataPoint.FlagsImmutable`
+  - `ExponentialHistogramDataPoint.Flags` -> `ExponentialHistogramDataPoint.FlagsImmutable`
+  - `SummaryDataPoint.Flags` -> `SummaryDataPoint.FlagsImmutable`
+  - `MetricDataPointFlags` -> `MetricDataPointFlagsImmutable`
+  - `NewMetricDataPointFlags` -> `MetricDataPointFlagsImmutable`
```

Comment:
Will we rename it back? We probably should mention that in the comments and here?

Reply:
Already done https://github.com/open-telemetry/opentelemetry-collector/pull/5999/files#diff-39b6dfa0a0fdf8e13fb63075018e16a009a905cfd91c451af47768bac6268e96R24